Source/Core/Core/AchievementManager.cpp

```diff
@@ -204,6 +204,19 @@ void AchievementManager::SetBackgroundExecutionAllowed(bool allowed)
     DoIdle();
 }
 
+void AchievementManager::CalculateHash(char hash[33], const std::string& file_path)
+{
+  rc_hash_filereader volume_reader{
+      .open = &AchievementManager::FilereaderOpenByFilepath,
+      .seek = &AchievementManager::FilereaderSeek,
+      .tell = &AchievementManager::FilereaderTell,
+      .read = &AchievementManager::FilereaderRead,
+      .close = &AchievementManager::FilereaderClose,
+  };
+  rc_hash_init_custom_filereader(&volume_reader);
+  rc_hash_generate_from_file(hash, RC_CONSOLE_GAMECUBE, file_path.c_str());
+}
```

Comment:
Please change the function signature to `std::string AchievementManager::CalculateHash(const std::string& file_path)` so that callers don't need to care about the array size.